pkg/graph/dag/dag.go

```diff
@@ -136,6 +136,84 @@ func (d *DirectedAcyclicGraph[T]) AddDependencies(from T, dependencies []T) erro
 	return nil
 }
 
+// TopologicalSortLevels returns the vertices of the graph grouped by topological levels
+// using Kahn's algorithm. Each level contains vertices that have no dependencies on each
+// other and can be processed in parallel. Within each level, vertices are sorted by their
+// original Order for stability.
+//
+// For example, given a graph:
+//
+//	A -> C
+//	B -> C
+//	C -> D
+//
+// The result would be: [[A, B], [C], [D]]
+// where A and B can be processed in parallel, then C, then D.
+func (d *DirectedAcyclicGraph[T]) TopologicalSortLevels() ([][]T, error) {
+	// Calculate in-degrees (number of dependencies) for each vertex
+	inDegree := make(map[T]int)
```

Comment:
currently  for each vertex in the current level, you scan all vertices to find dependents. This balloons runtime to O(V²) for dense cases. I dont think its a major issue but you can probably optimize for

```
reverse := make(map[T][]T)
for id, v := range d.Vertices {
    for dep := range v.DependsOn {
        reverse[dep] = append(reverse[dep], id)
    }
}

for _, vertex := range currentLevel {
    for _, dependentID := range reverse[vertex.ID] {
        inDegree[dependentID]--
        if inDegree[dependentID] == 0 {
            nextLevel = append(nextLevel, d.Vertices[dependentID])
        }
    }
}
```

That produces O(V + E) as intended